### PR TITLE
Implements destroy() method

### PR DESCRIPTION
**Pull request by Indigo744**
Finalize #117 and fix last leak in #118!

### CODE REVIEW

**Comment by Indigo744** [comment]
Wait for merging. There are problems with the UT and the way I destroy everything.
It seems that there is an issue with how we handle the resize event `$(window).on("resize")`.

**Comment by Indigo744** [comment]
Ok now it is working.
To explain a little bit, I had some problem because of the `resize` event attached on the window. I initially forgot to detach it in the destroy function, and thus it was kept between instance (and tests!).
In order to detach if using `off`, I needed to pass also the eventHandler callback to be sure I detach the correct one. Hence why the refactorization of the code regarding the map resizing.

File: js/jquery.mapael.js [hunk 1]
```diff
@@ -250,6 +236,75 @@
...
+         *     
+         * This method is also automatically called if the user try to call mapael
+         * on a container already containing a mapael instance
+         */
+        destroy: function() {
+            var self = this;
+            // Empty the container (this will also detach all event listeners)
+            self.$container.empty();
+            // Detach the global resize event handler
+            if (self.onResizeEvent) $(window).off("resize." + pluginName, self.onResizeEvent);
```
Comment (neveldo) [inline js/jquery.mapael.js]: Why not just call `$(window).off("resize." + pluginName);` ?

Comment (Indigo744) [inline js/jquery.mapael.js]: If we have two maps you will detach all events in that case :-)

Comment (neveldo) [inline js/jquery.mapael.js]: Yes, you are right, I didn't thought about that case :)

**Comment by neveldo** [comment]
Great work, thank you ! I merge your PR :)

**Pull request merged**